packages/sync/src/manager.ts

```diff
@@ -305,11 +307,49 @@ export function createSyncManager(): SyncManager {
 		return createPersistedCRDTDoc( entityState.ydoc );
 	}
 
+	/**
+	 * Update the last persisted timestamp, and the user who last persisted the document in the CRDT document state map.
+	 *
+	 * This can be used to send notifications to other peers about when the document
+	 * was last updated, and by whom.
+	 *
+	 * @param {ObjectType} objectType Object type.
+	 * @param {ObjectID}   objectId   Object ID.
+	 * @param {string}     origin     The source of change.
+	 */
+	function recordPersistence(
+		objectType: ObjectType,
+		objectId: ObjectID,
+		origin: string
+	): void {
+		const entityId = getEntityId( objectType, objectId );
+		const entityState = entityStates.get( entityId );
+
+		if ( ! entityState ) {
+			return;
+		}
+
+		const { ydoc } = entityState;
+
+		const lastPersistedAt = Date.now();
+
+		Y.transact(
+			ydoc,
+			() => {
+				const stateMap = ydoc.getMap( 'state' );
+				stateMap.set( PERSISTED_AT_KEY, lastPersistedAt );
+				stateMap.set( PERSISTED_BY_KEY, ydoc.clientID );
```

Comment:
I've been wondering if this is really the best approach. It's not really document state and it feels fragile.

Reply:
The problem I guess is that, we need to be able to tell the other clients that the document has been updated and by whom without spamming notifications. 

If we hook into each change, we'd be spamming notifications. If we hook into when the post is saved, then those changes are batched at least.

That's why I stuck with this approach.

Reply:
I realized why I like this approach. It's like our very own redux action that we can fire that plugins can hook into to react the way they want.